File: 002/002.py
# Calculates sum of even numbers in Fibonacci sequence under 4 million

def fibonacci(n):
  if n == 1:
    return 1
  elif n == 2:
    return 2
  else:
    return fibonacci(n-1) + fibonacci(n-2)

sum = 0
x = 1
while fibonacci(x) <= 4000000:
  if fibonacci(x)%2 == 0:
    sum = sum+fibonacci(x)
  x+=1

print sum

File: 002/002.cpp
#include<iostream>
using namespace std;

int fibonacci(int n)
{
  if(n==1) return 1;
  if(n==2) return 2;
  return fibonacci(n-1) + fibonacci(n-2);
}

int main(int argc, char* argv[])
{
  int sum = 0;
  for(int i=1; fibonacci(i)<=4000000; i++)
  {
    if(fibonacci(i)%2 == 0)
    {
      sum += fibonacci(i);
    }
  }
  cout << sum << endl;
  return 0;
}
